gnome-initial-setup/pages/language/cc-language-chooser.c

```diff
@@ -307,12 +308,90 @@ language_visible (GtkListBoxRow *row,
         return visible;
 }
 
+/*
+ * Copied from tracker/src/libtracker-fts/tracker-parser-glib.c under the GPL
+ * And then from gnome-shell/src/shell-util.c
+ *
+ * Originally written by Aleksander Morgado <aleksander@gnu.org>
+ */
+#define IS_SOFT_HYPHEN(c) ((c) == 0x00AD)
+#define IS_CDM_UCS4(c) (((c) >= 0x0300 && (c) <= 0x036F)  || \
+                        ((c) >= 0x1DC0 && (c) <= 0x1DFF)  || \
+                        ((c) >= 0x20D0 && (c) <= 0x20FF)  || \
+                        ((c) >= 0xFE20 && (c) <= 0xFE2F))
+
+static gchar*
+normalize_casefold_and_unaccent (const gchar *str)
```

Comment:
This code already exists in cc-util.c in this directory, so just #include "cc-util.h" instead of copying it here.

P.S. If I didn't ask you to remove this code, I would have asked you to reindent it to match the rest of the file. Good to do when copying from another project.